### DISCUSSION TITLE
Install on RN 0.70

### DISCUSSION THREAD

**Discussion by carlosriveroib** [invertase/react-native-firebase#6536] [Q&A]
Does somebody know how to install on RN 0.70? I'm trying for 2 days and I cannot. I appreciate some help. Thank you very much

**Comment by mikehardy** [comment]
Hi there! Sure, like this:

https://github.com/mikehardy/rnfbdemo/blob/rn70-hermes-use-frameworks/make-demo.sh

That's a demonstration start to finish - there was not that much that changed between rn69 / rn70 as far as I know? I still have issues with Hermes on iOS and have not had time to run them down, but if you disable flipper and disable hermes it works the same as rn69, with the exception of a bunch of things needed for macCatalyst go away.

I'm testing something with Xcode 14 that looks problematic and used to only be needed for macCatalyst builds - it's related to code-signing and did not happen for non-macCatalyst on Xcode 13 but now appears necessary for all builds on Xcode 14.

Specifically this:
https://github.com/mikehardy/rnfbdemo/blob/058dc304ca739f40698225753c12164875af1907/make-demo.sh#L263

Which came from this https://github.com/CocoaPods/CocoaPods/issues/8891#issuecomment-573301570

But now this appears to be needed when you archive (debug builds are fine):
https://github.com/CocoaPods/CocoaPods/issues/8891#issuecomment-1201465446

**Reply by carlosriveroib** [reply]
I really appreciate that, but I'm looking for a step by step guide or help. That bash file seems hard to read and specially for someone with less experience.

Seems like it's doing things that I don't need.
For example.
It's using npx @react-native-community/cli init rnfbdemo --skip-install --version=0.70.0-rc.3
I want to use npx react-native init with a typescript template

Could we elaborate a guide together?

**Reply by mikehardy** [reply]
Sorry, the bash file *is* the step by step guide we've got, the comments are all quite specific so it may be hard to read but I think that is also a bit of an irreducible complexity

The template you are using should not matter, it's only altering things with regard to native code, and primarily with Podfile

If you propose a PR to the make-demo.sh script we can collaborate that way, but the things you don't need are all needed by other people  so I will not remove them - the PR should mostly be aimed at clarifying comments as needed (that would be a big help!) and updating it to rn 0.70.0 + Xcode compatibility (specifically, changing that code signing thing for pod resources to be used in general and to say "CODE_SIGNING_ALLOWED" = no, as linked in the comment. I have just successfully uploaded a test flight build with that.

You haven't provided any sort of errors that would indicate what your build is failing on so for me, I stick *very* closely to that script since there are an uncountable number of ways to incorrectly build things, but that script builds it right *every* time.
